internal/middleware/reverse_proxy.go

```diff
@@ -6,11 +6,14 @@ import (
 
 	"github.com/pomerium/pomerium/internal/cryptutil"
 	"github.com/pomerium/pomerium/internal/log"
+	"github.com/pomerium/pomerium/internal/telemetry"
 )
 
 func SignRequest(signer cryptutil.JWTSigner, id, email, groups, header string) func(next http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, span := telemetry.StartSpan(r.Context(), "middleware.SignRequest")
```

Comment:
Should this be keeping the returned context and calling r.WithContext() when we ServeHTTP on the next middleware?